app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt

```diff
@@ -556,19 +558,36 @@ open class FileListFragment : MultiSelectFragment(MATOMO_CATEGORY), SwipeRefresh
 
     private fun observeOfflineDownloadProgress() {
         mainViewModel.observeDownloadOffline(requireContext().applicationContext).observe(viewLifecycleOwner) { workInfoList ->
-            if (workInfoList.isEmpty()) return@observe
+            updateFileStatus(workInfoList, DownloadWorker.FILE_ID, DownloadWorker.PROGRESS)
+        }
 
-            val workInfo = workInfoList.firstOrNull { it.state == WorkInfo.State.RUNNING }
+        mainViewModel.updateVisibleFiles.observe(viewLifecycleOwner) {
+            updateVisibleProgresses()
+        }
+    }
 
-            if (workInfo == null) {
-                updateVisibleProgresses()
-                return@observe
+    private fun observeOfflineBulkDownloadProgress() {
+        mainViewModel.observeBulkDownloadOffline(requireContext().applicationContext)
+            .observe(viewLifecycleOwner) { workInfoList ->
+                updateFileStatus(workInfoList, BulkDownloadWorker.FILE_ID, BulkDownloadWorker.PROGRESS)
             }
+    }
+
+    private fun updateFileStatus(workInfoList: List<WorkInfo>, fileIdKey: String, progressKey: String) {
+        if (workInfoList.isEmpty()) return
+
+        val workInfos = workInfoList.filter { it.state == WorkInfo.State.RUNNING }
+
+        if (workInfos.isEmpty()) {
+            updateVisibleProgresses()
+            return
+        }
 
-            val fileId: Int = workInfo.progress.getInt(DownloadWorker.FILE_ID, 0)
-            if (fileId == 0) return@observe
+        workInfos.forEach { workInfo ->
```

Comment:
Why a foreach ?

Reply:
Yes you're right, I only receive 1 `WorkInfo` every time here.